src/main/java/com/hms/flexyosisoftconnector/dataserver/OSIsoftServer.java

```diff
@@ -76,56 +79,82 @@ public class OSIsoftServer {
   /** Constructs the OSIsoftServer object. */
   public OSIsoftServer() {}
 
+  /** HTTP Method string for Post requests */
+  static final String HTTP_METHOD_POST = "POST";
+
+  /** HTTP Method string for Get requests */
+  static final String HTTP_METHOD_GET = "GET";
+
+  /**
+   * Repeated Exception handling code for HTTP requests
+   *
+   * @param e Exception
+   * @param ErrorMessage Specific message to log
+   */
+  private static void RequestHttpsError(Exception e, String ErrorMessage) {
+    connected = false;
+    Logger.LOG_CRITICAL(ErrorMessage);
+    Logger.LOG_EXCEPTION(e);
+  }
+
   /**
    * Makes an HTTPS request against the OSIsoft server and validates the response code.
    *
-   * @param CnxParam Request parameters
+   * @param CnxParam Request parameters: GET or POST
    * @param Method Request method
    * @param Headers Request headers
    * @param TextFields Request Payload
    * @param FileName File name for response
-   * @return The HTTP response code
+   * @return True/False message was successfully sent
+   * @throws IllegalArgumentException

```

Comment:
Why does it throw an `IllegalArgumentException`?

Reply:
If CnxParam is not "GET" or "POST", it is an illegal argument. This is all that the extension library will support.  This exception should only be thrown during development when a developer tries a "PUT", etc.

Reply:
The reason should be written in the Javadoc.

```suggestion
   * @throws IllegalArgumentException if the specified parameter, {@code CnxParam}, is not an allowed value
```

Reply:
Add missing Javadoc description for exception